core/src/main/java/org/mule/config/pool/DefaultThreadPoolFactory.java

```diff
@@ -81,8 +84,9 @@ private void configureThreadPoolExecutor(String name, ThreadingProfile tp, Threa
     @Override
     public ScheduledThreadPoolExecutor createScheduledPool(String name, ThreadingProfile tp)
     {
-        ScheduledThreadPoolExecutor pool = internalCreateScheduledPool(tp);
-        configureThreadPoolExecutor(name, tp, pool);
+        ThreadingProfile threadingProfileCopy = tp instanceof DirectThreadingProfile ? tp : new ImmutableThreadingProfile(tp);
```

Comment:
No, thats whats so horrible about ImmutableThreadingProfie, it doesn't wrap/delegate, rather it copies fields and then provides it's own implementation for createPool, createWorkManager etc.